### PR TITLE
Cron job

### PR DESCRIPTION
**Pull request by GilDavidi**

### CODE REVIEW

File: cronJob/cronJob.js [hunk 1]
```diff
@@ -1,58 +1,56 @@
...
-                let today= moment();
-                if(next_date.format('YYYY-MM-DD') == today.format('YYYY-MM-DD'))
-                {
-                   addMonthForDate(subscription);
+        subscriptions.forEach(subscription => {
+                let next_date = moment(subscription.next_date);
+                let today = moment();
+                if (next_date.format('YYYY-MM-DD') == today.format('YYYY-MM-DD')) {
 
+                    updateSubscription(subscription);
```
Comment (itay45977) [inline cronJob/cronJob.js:48]: Great that you've decided to send the whole sub

File: cronJob/cronJob.js [hunk 2]
```diff
@@ -1,58 +1,56 @@
...
+const updateSubscription = (subscription, credits) => {
 
     let nextMonth = moment().add(1, 'M').format('YYYY-MM-DD HH:mm:ss');
     let updatedSubscription = new Subscription;
     updatedSubscription = subscription;
     updatedSubscription.next_date = nextMonth;
-    console.log("updatedSubscription:", updatedSubscription)
 
-    Subscription.findByIdAndUpdate(subscription._id, updatedSubscription, {new: true}, (err, docs) => {
+    Subscription.findByIdAndUpdate(subscription._id, updatedSubscription, {new: true}, (err) => {
```
Comment (itay45977) [inline cronJob/cronJob.js:18]: Nice command, I use it a lot :)

File: cronJob/cronJob.js [hunk 3]
```diff
@@ -1,58 +1,56 @@
...
+const sendQueueUpdate = require("../RMQ/rabbitMQ");
 
-require("dotenv").config({path: '.env'});
+const logger = serverLogger.log;
 
-const addMonthForDate =(subscription) => {
+
+const updateSubscription = (subscription, credits) => {
 
     let nextMonth = moment().add(1, 'M').format('YYYY-MM-DD HH:mm:ss');
```
Comment (itay45977) [inline cronJob/cronJob.js:13]: this command is not familiar to me - great resrearch about it

File: cronJob/cronJob.js [hunk 4]
```diff
@@ -1,58 +1,56 @@
...
 
-    Subscription.findByIdAndUpdate(subscription._id, updatedSubscription, {new: true}, (err, docs) => {
+    Subscription.findByIdAndUpdate(subscription._id, updatedSubscription, {new: true}, (err) => {
         if (err) {
-            console.log(err)
+            logger.error(`findByIdAndUpdate failed: ${err} to user email: ${subscription.email}`);
+        } else {
+            logger.info(`Next date subscription updated successfully to user email: ${subscription.email}`);
+
+            sendQueueUpdate(subscription.email, credits);
```
Comment (itay45977) [inline cronJob/cronJob.js:24]: Please send the plan to the queue, its the most important data about the user

File: RMQ/rabbitMQ.js [hunk 5]
```diff
@@ -4,19 +4,21 @@ const serverLogger = require(`../logger`);
...
+            const stringMsg = JSON.stringify(subscriptionDetails);
             ch.assertQueue(q, {durable: false});
-            setInterval(() => {
-                ch.sendToQueue(q, Buffer.from(stringMsg))
-                    .then(logger.info(`RMQ message sent: ${msg}`));
-            })
+            const sendQueue = async () => {
+                ch.sendToQueue(q, Buffer.from(stringMsg));
+            };
+            await sendQueue().then(() => logger.info(`RMQ- update free plan message sent to user: ${user_mail} with: ${credits} credits`));
```
Comment (itay45977) [inline RMQ/rabbitMQ.js]: It you write 'free' static, so this function is for the cron job only, right? 
we'll add another function to the pro/premium

**Review by itay45977 — APPROVED** [review]
Great job, looks awesome.
Just let me know if the queue's function is for the free and cronjob only

File: RMQ/rabbitMQ.js [hunk 6]
```diff
@@ -4,19 +4,21 @@ const serverLogger = require(`../logger`);
...
                 "credits": {credits}
             };
-            const stringMsg = JSON.stringify(msg);
+            const stringMsg = JSON.stringify(subscriptionDetails);
             ch.assertQueue(q, {durable: false});
-            setInterval(() => {
-                ch.sendToQueue(q, Buffer.from(stringMsg))
-                    .then(logger.info(`RMQ message sent: ${msg}`));
-            })
+            const sendQueue = async () => {
```
Comment (DaveAvigad) [inline RMQ/rabbitMQ.js]: `async` here is redundant

File: RMQ/rabbitMQ.js (hunk 5 above)
Comment (DaveAvigad) [inline RMQ/rabbitMQ.js]: you mix between async and Promises. decide either go async or promises not both
you can't do `await sendQueue().then()`

File: controllers/subscriptionController.js [hunk 7]
```diff
@@ -1,12 +1,21 @@
 const {Subscription} = require('../models/subscriptions')
```
Comment (DaveAvigad) [inline controllers/subscriptionController.js:1]: seems that you do not use eslint in your project.
please fix ASAP!!

File: controllers/subscriptionController.js [hunk 8]
```diff
@@ -1,12 +1,21 @@
...
-
+const serverLogger = require(`../logger`);
+const logger = serverLogger.log;
 
 module.exports = {
     getAllPSubscription: (req, res) => {
         Subscription.find({}).then(result => {
```
Comment (DaveAvigad) [inline controllers/subscriptionController.js]: FYI this 
```
 Subscription.find({}).then(result => {
            res.send(result);
        })
            .catch(err => logger.error(err));
```
can be written in shorter - 
``` 
Subscription
  .find({})
  .then(res.send)
  .catch(logger.error);
```

File: cronJob/cronJob.js [hunk 9]
```diff
@@ -1,58 +1,56 @@
 const cron = require('node-cron');
-const subscriptionController = require('../controllers/subscriptionController');
 const request = require('request');
 const moment = require("moment");
-const {format} = require("winston");
 const {Subscription} = require('../models/subscriptions')
+const serverLogger = require(`../logger`);
+const sendQueueUpdate = require("../RMQ/rabbitMQ");
```
Comment (DaveAvigad) [inline cronJob/cronJob.js:6]: please keep the convention. file names should be in `camelCase`d.

File: cronJob/cronJob.js (hunk 3 above)
Comment (DaveAvigad) [inline cronJob/cronJob.js:13]: if you use this pattern `'YYYY-MM-DD HH:mm:ss'` in other places please consider moving it into a constant

File: cronJob/cronJob.js [hunk 10]
```diff
@@ -1,58 +1,56 @@
...
     })
 }
 
 
-
-
-
 const startCronJob = () => {
     let subscriptions;
     //code for 02 at night -cron.schedule("00 02 * * *", ()
```
Comment (DaveAvigad) [inline cronJob/cronJob.js:33]: do not commented-out code to production!

File: cronJob/cronJob.js [hunk 11]
```diff
@@ -1,58 +1,56 @@
...
 
-    cron.schedule("*/10 * * * * *", () => {
+    cron.schedule("*/6 * * * * *", () => {
         //get all subscriptions
-        request('http://localhost:'+process.env.PORT +'/subscription', {json: true}, (err, res, body) => {
+        request('http://localhost:' + process.env.PORT + '/subscription/getAllSubscriptionByName/Free', {json: true}, (err, res, body) => {
```
Comment (DaveAvigad) [inline cronJob/cronJob.js:37]: remember that the host name should be a dynamic value. please move to config/ env var

Comment (DaveAvigad) [inline cronJob/cronJob.js:37]: why use `request` and not `Axios` package?

**Review by DaveAvigad — APPROVED** [review]
nice job!
left some comments.
please fixed and merge.
**approved.**

**Pull request merged**